### PR TITLE
Handle file access exception while looking for plugins

### PR DESCRIPTION
**Pull request by nicolas-f**
My linux antivirus is raising an exception if you try to retrieve filesystem data on its binary:

/usr/bin/sentinelctl

So I updated the code to handle FileSystem exception while looking for pluings in PATH.

### CODE REVIEW

**Comment by nicolas-f** [comment]
About #241

File: src/plugin/plugin.cpp [hunk 1]
```diff
@@ -58,56 +58,66 @@ namespace linuxdeploy {
...
+                                }
                             }
+                        } catch (const PluginError& e) {
+                            ldLog() << LD_WARNING << "Could not load plugin" << i->path() << LD_NO_SPACE << ": " << e.what();
                         }
-                    } catch (const PluginError& e) {
-                        ldLog() << LD_WARNING << "Could not load plugin" << i->path() << LD_NO_SPACE << ": " << e.what();
                     }
+                } catch(std::filesystem::filesystem_error const& ex) {
+                    ldLog() << LD_WARNING
```
Comment (TheAssassin) [inline src/plugin/plugin.cpp:114]: Should be `LD_DEBUG`, I guess. Please fix the indentation to match the other code.

File: src/plugin/plugin.cpp [hunk 2]
```diff
@@ -58,56 +58,66 @@ namespace linuxdeploy {
...
+                    << "Got file system error while looking for plugins :  " << ex.what() << '\n'
```
Comment (TheAssassin) [inline src/plugin/plugin.cpp:115]: Please use `std::endl`.

**Review by TheAssassin — CHANGES_REQUESTED** [review]
Please see my comments.

Do we really have to wrap so much code? I assume the exception occurs in the first few lines of probing code.

**Comment by nicolas-f** [comment]
I reduce it and can't reduce it further. The exception is raised even in the call of fs::is_directory

Thanks

File: src/plugin/plugin.cpp [hunk 3]
```diff
@@ -37,7 +37,7 @@ namespace linuxdeploy {
 
         std::map<std::string, IPlugin*> findPlugins() {
             std::map<std::string, IPlugin*> foundPlugins;
-
+
```
Comment (TheAssassin) [inline src/plugin/plugin.cpp:40]: Please clean up this whitespace. Otherwise, this can be merged.

**Review by TheAssassin — APPROVED** [review]

**Comment by TheAssassin** [comment]
Build failure is unrelated to your changes. Merging. Thanks!

**Pull request merged**